datalad/downloaders/providers.py

```diff
@@ -327,9 +327,15 @@ def get_provider(self, url, only_nondefault=False, return_all=False):
         matching_providers = []
         for provider in self._providers[::-1]:
             for url_re in provider.url_res:
-                if re.match(url_re, url):
-                    lgr.debug("Returning provider %s for url %s", provider, url)
-                    matching_providers.append(provider)
+                try:
+                    if re.match(url_re, url):
+                        lgr.debug("Returning provider %s for url %s", provider, url)
+                        matching_providers.append(provider)
+                except Exception:
```

Comment:
isn't exception more specific? but oh my:
```
$> python -c "import re; re.match('(?<ZZ>.*)', '1')"   
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
  File "/usr/lib/python2.7/re.py", line 251, in _compile
    raise error, v # invalid expression
sre_constants.error: syntax error
```
remnants of the past syntax (pre 2.5?)

Overall, here it is - `re.error`
```
import re; 

try:
  re.match('(?<ZZ>.*)', '1')
except re.error as exc:
  print(type(exc))
```

Please add a unittest to `datalad/downloaders/tests/test_providers.py` to assure that no exception is raised and the warning is logged ( `with swallow_logs(new_level=logging.WARNING) as cml: ...assert_in(...)` - search for those to see how it is done)

Reply:
> isn't exception more specific? but oh my

The reason I just caught all exceptions is that I can't think of any type of exception that would fail to compile the regex that you wouldn't want to turn into a warning. Regardless of how it's invalid, it should probably warn.

> Please add a unittest to [...]

Sure.

Reply:
If it is just a wrong regex - it is one thing.  But may be it managed to run out of RAM and issued MemoryError right then -- should we just keep plowing forward?  If user preseed Ctrl-C - should we ignore and just skip this particular regex issuing a warning that regex was invalid? ;-)
Regardless how unlikely, unless there is really an intention to ignore all errors, better to be specific.  If something else unexpected happens (e.g. `KeyboardInterrupt`) - it might need to be handled "upstairs".